modules/metastorage/src/main/java/org/apache/ignite/internal/metastorage/impl/MetaStorageRaftGroupEventsListener.java

```diff
@@ -156,11 +178,16 @@ private void executeIfLeader(OnLeaderAction action) {
     }
 
     private CompletableFuture<Void> executeIfLeaderImpl(OnLeaderAction action) {
+        return executeWithStatus((service, term, isLeader) -> action.apply(service, term));
```

Comment:
It seems that here it is necessary to return `completedFuture(null)` if not the leader.

Reply:
What do you mean?